### PR TITLE
Better parallelize CI jobs

### PR DESCRIPTION
**Pull request by Earlopain**
Drop appraisal while we're at it. Makes it a bit easier

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -186,13 +174,13 @@ jobs:
         uses: actions/upload-artifact@v4
         if: failure()
         with:
-          name: screenshots
+          name: screenshots-${{ matrix.pg }}-${{ matrix.gemfile }}-${{ matrix.ruby }}
```
Comment (bensheldon) [inline .github/workflows/test.yml:177]: you fixed a longstanding annoyance! 🙌🏻

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -86,22 +86,39 @@ jobs:
           path: demo/log
 
   test:
-    name: Test
+    name: Test - Ruby ${{ matrix.ruby }} - Postgres ${{ matrix.pg }} - ${{ matrix.gemfile }}
```
Comment (bensheldon) [inline .github/workflows/test.yml:89]: I think Rails is likely to be the culprit in most matrix-related failures:

```suggestion
    name: Test - ${{ matrix.gemfile }} - Ruby ${{ matrix.ruby }} - PG ${{ matrix.pg }}
```

**Review by bensheldon — APPROVED** [review]
💯 🥳 🥇 

This is amazing!!!! THANK YOU!! You cleaned up so many problems and it's so much better now (and TIL `eval_gemfile`).  

I'm a little surprised that there aren't more gem tweaks that are necessary because I felt like I was frequently struggling with Appraisal, so this is great!

**Comment by bensheldon** [comment]
btw, that failure with Rails 6.1 is sorta known: https://github.com/bensheldon/good_job/issues/1280

It's fixed by pulling from the Rails 6.1-stable git branch, though I'm a little surprised it shows up now rather than before.

**Comment by Earlopain** [comment]
> btw, that failure with Rails 6.1 is sorta known:

6.1 previously only ran with Ruby 3.0 which explains why it didn't pop up before. I could add excludes to the matrix to preserve that but just pulling 6-1-stable seems fine for CI. WDYT?

**Comment by Earlopain** [comment]
For `eval_gemfile`, it sorta is/isn't actually endorsed for usage by the bundler maintainers https://github.com/rubygems/rubygems/pull/6292#issuecomment-1752938643 https://github.com/rubygems/bundler-site/pull/1081. It makes things so very simple though and don't believe it'll disappear (and if it does, let's just handle that then)

(I'm hugely surprised that this PR turned out well with just one try. Usually my interactions with github actions require much more force-pushes :upside_down_face:)

**Comment by bensheldon** [comment]
🫶 thank you!

**Pull request merged**